gammapy/datasets/core.py

```diff
@@ -331,6 +331,12 @@ def to_spectrum_datasets(self, region):
 
         return datasets
 
+    def to_asimov_datasets(self):
+        """Create Asimov datasets from the current models."""
+        return Datasets(
+            [d._to_asimov_dataset(name=d.name, models=self.models) for d in self]
+        )
+
```

Comment:
Usually you want to create the asimov dataset for a specific alternative hypothesis. 
I think there are two approaches:
1. Set the desired hypo first for the model attached to the og datasets, then create the asimov dataset and restore the parameters on the original dataset (requires using context manager for user to avoid refitting)
2. Make it possible to give this method a parameters set to create the dataset from

I like this second one better, because you directly get the hypo asimov dataset without touching the original dataset

What do you think?

Reply:
One can always do
```
dataset.models = models1
dataset_asimov = dataset.to_asimov_dataset()
ts1 = dataset_asimov.stat_sum()

dataset.models = models2
...
```

but maybe 
```
dataset_asimov = dataset.to_asimov_dataset(models)
```
can be simpler...

Reply:
The hypothesis testing can be defined in dependently of the datasets:
`test = TestStatisticNested([parameter], [null_value])`
then one can get :
`ts_asimov = test.ts_asimov(datasets)`